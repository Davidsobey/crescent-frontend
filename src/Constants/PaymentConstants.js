const PaymentConstants = {

  GETALL_REQUEST: 'GETALL_PAYMENT_REQUEST',
  GETALL_SUCCESS: 'GETALL_PAYMENT_SUCCESS',
  GETALL_FAILURE: 'GETALL_PAYMENT_FAILURE',

  GETBYID_REQUEST: 'GETBYID_PAYMENT_REQUEST',
  GETBYID_SUCCESS: 'GETBYID_PAYMENT_SUCCESS',
  GETBYID_FAILURE: 'GETBYID_PAYMENT_FAILURE',

  CHANGE_PAYMENT_STATUS_REQUEST: 'CHANGE_PAYMENT_STATUS_REQUEST',
  CHANGE_PAYMENT_STATUS_SUCCESS: 'CHANGE_PAYMENT_STATUS_SUCCESS',
  CHANGE_PAYMENT_STATUS_FAILURE: 'CHANGE_PAYMENT_STATUS_FAILURE',
};

export default PaymentConstants;
