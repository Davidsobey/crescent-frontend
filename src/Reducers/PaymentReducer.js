import PaymentConstants from '../Constants/PaymentConstants';
import UserConstants from '../Constants/UserConstants';

function PaymentReducer(state = {}, action) {
  switch (action.type) {
    case PaymentConstants.GETALL_REQUEST:
      return {
        loading: true,
      };
    case PaymentConstants.GETALL_SUCCESS:
      return Object.assign({}, state, {
        paymentStatuses: action.paymentStatuses,
        loading: false,
      });
    case PaymentConstants.GETALL_FAILURE:
      return {
        paymentStatuses: {},
        loading: false,
      };
    case PaymentConstants.CHANGE_PAYMENT_STATUS_REQUEST:
      return Object.assign({}, state, {
        paymentStatusChanging: true,
      });
    case PaymentConstants.CHANGE_PAYMENT_STATUS_SUCCESS:
      return Object.assign({}, state, {
        subscription: undefined,
        paymentStatusChanging: false,
      });
    case PaymentConstants.CHANGE_PAYMENT_STATUS_FAILURE:
      return Object.assign({}, state, {
        error: action.error,
        paymentStatusChanging: false,
      });
    case UserConstants.LOGOUT:
      return {};
    default:
      return state;
  }
}

export default PaymentReducer;
