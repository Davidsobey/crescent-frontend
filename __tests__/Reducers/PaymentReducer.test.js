// Reducer to be tested
import PaymentReducer from '../../src/Reducers/PaymentReducer';
import PaymentConstants from '../../src/Constants/PaymentConstants';

describe('PaymentReducer', () => {
  test('GETALL_SUCCESS', () => {
    const action = { 
      type: PaymentConstants.GETALL_SUCCESS, 
      paymentStatuses: [],
    };

    expect(PaymentReducer(undefined, action)).toMatchSnapshot();
  });
});
