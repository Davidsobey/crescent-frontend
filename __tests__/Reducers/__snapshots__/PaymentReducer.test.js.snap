// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentReducer GETALL_SUCCESS 1`] = `
Object {
  "loading": false,
  "paymentStatuses": Array [],
}
`;
